test/suites/pki.sh

```diff
@@ -106,6 +107,58 @@ test_pki() {
     lxc_remote remote remove pki-lxd
 
 
+    ### Restricted CA signed client certificate with `core.trust_ca_certificates` disabled.
+
+    # Set up the client config
+    cp "${TEST_DIR}/pki/keys/restricted.crt" "${LXD_CONF}/client.crt"
+    cp "${TEST_DIR}/pki/keys/restricted.key" "${LXD_CONF}/client.key"
+    cat "${LXD_CONF}/client.crt" "${LXD_CONF}/client.key" > "${LXD_CONF}/client.pem"
+
+    # Try adding remote using an incorrect token. This should fail even though the client certificate
+    # has been signed by the CA because `core.trust_ca_certificates` is not enabled.
```

Comment:
Unless I missed it, setting `core.trust_ca_certificates=true` and trying the `remote add` is not tested, only "token-based remote add" is. Worth adding?

Reply:
This is tested [here](https://github.com/canonical/lxd/pull/13702/commits/8f795403bbcdc59e64a0f752a7d473321d4b21aa#diff-44dd5a04f4d990338c795090d83d906e54ddf955eadc62ec51483a6586061151R162-R220) I believe.

Reply:
Indeed, thanks for the pointer.